### PR TITLE
instance.report_runless_asset_event

### PR DESCRIPTION
**Pull request by alangenfeld**
adds an instance method for reporting runless asset events

## How I Tested These Changes

added test

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16520** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16520" target="_blank">Graphite</a> 
    * **PR #16571** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16571" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16571?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/instance/__init__.py [hunk 1]
```diff
@@ -2820,3 +2835,41 @@ def get_latest_materialization_code_versions(
                 result[asset_key] = data_provenance.code_version if data_provenance else None
 
         return result
+
+    @experimental
+    def report_runless_asset_event(
```
[the remark this answers is not in the document]
Comment (schrockn) [reply python_modules/dagster/dagster/_core/instance/__init__.py:2840]: Ah great. Yeah "runless" feels like a less heavy concept in the system. This is just directly what it is.

**Comment by johannkm** [comment]
Currently this won't let you report a failure right? One more case where a MATERIALIZATION_FAILED event would be nice

**Comment by johannkm** [comment]
There's a couple places in the product (e.g. asset status) where we do joins on materializations and runs, may need an update

**Pull request merged**